submissions/HannaSyn/oop-exercise/index.js

```diff
@@ -0,0 +1,88 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Inhabitant {
+   constructor ( name, gender, friends, say, biologicalSpecies ) {
+      this.name = name;
+      this.biologicalSpecies = biologicalSpecies;
+      this.gender = gender;
+      this.friends = friends;
+      this.say = say
+   }
+   speak() {
+      return `${this.say} I'm ${this.biologicalSpecies}! My name is ${this.name}, my gender is ${this.gender}, and I have friends: ${this.friends}.`;
+   }
+}
+
+class Human extends Inhabitant {
+   constructor( name, gender, friends, say, legs, hands ) {
+      super( name, gender, friends, say, 'human');
+      this.legs = legs;
+      this.hands = hands;
+   } 
+   speak() {
+      return super.speak() + ` I have ${this.legs} legs and ${this.hands} hands.`
+   }
+}
+
+class Animal extends Inhabitant {
+   constructor( name, gender, friends, say, biologicalSpecies, paws) {
+      super( name, gender, friends, say, biologicalSpecies )
+      this.paws = paws;
+   }
+   speak() {
+      return super.speak() + ` I have ${this.paws} paws.`
+   }
+}
+
+class Cat extends Animal {
+   constructor( name, gender, friends, say, paws) {
+      super( name, gender, friends, say, 'cat', paws );
+   } 
+}
+
+class Dog extends Animal {
+   constructor( name, gender, friends, say, paws, Human ) {
+      super( name, gender, friends, say, 'dog', paws);
+      this.master = Human;
+   } 
+   speak() {
+      return super.speak() + ` I love my master ${this.master.name} very much!`
+   }
+}
+
+
+const woman = new Human('Hanna', 'female', ['Vlad', 'Ernest', 'Emilia'], 'Hi!', 2, 2);
+const man = new Human('Vlad', 'male', ['Hanna', 'Ernest', 'Emilia'], 'Hello!', 2, 2);
+const cat = new Cat('Ernest', 'male', ['Vlad', 'Hanna', 'Emilia'], 'Meow!', 4);
+const dog = new Dog('Emilia', 'female', ['Vlad', 'Ernest'], 'Wooof!', 4, woman);
```

Comment:
Well, here friends list is just a set strings. What if want to send over real inhabitants as friends? This would give us liberty to describe the friends as we want them not limiting ourselves to names only.
Just like you did with a master for a dog.